chain/validation/state.go

```diff
@@ -239,3 +276,33 @@ type directStorage struct {
 func (d *directStorage) Get(cid cid.Cid) ([]byte, error) {
 	panic("implement me")
 }
+
+func fromActorCode(code vstate.ActorCodeCid) (cid.Cid, error) {
+	switch code {
+	case vstate.AccountActorCodeCid:
+		return actors.AccountActorCodeCid, nil
+	case vstate.StorageMinerCodeCid:
+		return actors.StorageMinerCodeCid, nil
+	case vstate.MultisigActorCodeCid:
+		return actors.MultisigActorCodeCid, nil
+	case vstate.PaymentChannelActorCodeCid:
+		return actors.PaymentChannelActorCodeCid, nil
+	default:
+		return cid.Undef, errors.Errorf("Unknown actor code: %v", code)
+	}
+}
+
+func fromSingletonAddress(addr vstate.SingletonActorAddress) (vstate.Address, error) {
+	switch addr {
+	case vstate.InitAddress:
+		return vstate.Address(actors.InitActorAddress.Bytes()), nil
+	case vstate.NetworkAddress:
+		return vstate.Address(actors.NetworkAddress.Bytes()), nil
+	case vstate.StorageMarketAddress:
+		return vstate.Address(actors.StorageMarketAddress.Bytes()), nil
+	case vstate.BurntFundsAddress:
+		return vstate.Address(actors.BurntFundsAddress.Bytes()), nil
+	default:
+		return "", errors.Errorf("Unknown singleton actor address: %v", addr)
+	}
+}
```

Comment:
For validation code (as opposed to production code) I'm ok panicking on programmer errors like this.